### DISCUSSION TITLE
Is deep metric learning perform better than traditional method (MLP) in few shot learning scenario?

### DISCUSSION THREAD

**Discussion by 117CCC** [KevinMusgrave/pytorch-metric-learning#418] [General]
First of all, thanks for this excellent library and it really helps.

I was told metric learning was very helpful for few shot learning problem, but I don't know if my coding logic is correct ,because I got different result in experiment. 

The dataset I use is an open source plant species dataset and total of 30 categories, each category 50 images.

1. Using the resnet50(pretrained = True) to process the dataset, highest accuracy is about 96%

2. Using the resnet50(pretrained = True)  as backbone to form the siamese network (positive pair and negative pair) to process the dataset, highest accuracy is about 90%

It seems that siamese network cannot perform better than single CNN even in very small size dataset.

I have some thoughts

1. 50 images for each category is not small a size dataset, probably I should use 5 images for each category.

2. Probably use pretrained model is not fair for metric learning method.

Welcome to discuss.

**Comment by KevinMusgrave** [comment]
> I was told metric learning was very helpful for few shot learning problem, but I don't know if my coding logic is correct ,because I got different result in experiment.

I've heard this type of thing too, but I don't know if there is real evidence to back it up. 

> 1. Using the resnet50(pretrained = True) to process the dataset, highest accuracy is about 96%

So you used the cross entropy loss for training?

> 2. Probably use pretrained model is not fair for metric learning method.

Well in pretty much every application, people will start with a pretrained model.
